features/environment.py

```diff
@@ -149,21 +151,29 @@ def create_trusty_uat_lxd_image(context: Context) -> None:
     :param context:
         A `behave.runner.Context`; this will have `image_name` set on it.
     """
-    now = datetime.datetime.now()
-    context.image_name = "behave-image-" + now.strftime("%s%f")
-    build_container_name = "behave-image-build-" + now.strftime("%s%f")
+    if context.container_to_run:
+        print(
+            "It is going to run the test in the existent container: ",
+            context.container_to_run,
+        )
+    else:
+
+        now = datetime.datetime.now()
+        context.image_name = "behave-image-" + now.strftime("%s%f")
+        build_container_name = "behave-image-build-" + now.strftime("%s%f")
 
-    launch_lxd_container(context, "ubuntu:trusty", build_container_name)
-    _install_uat_in_container(build_container_name)
-    _capture_container_as_image(build_container_name, context.image_name)
+        launch_lxd_container(context, "ubuntu:trusty", build_container_name)
+        _install_uat_in_container(build_container_name)
+        _capture_container_as_image(build_container_name, context.image_name)
 
     def image_cleanup() -> None:
         if context.config.image_clean:
             subprocess.run(["lxc", "image", "delete", context.image_name])
         else:
             print("Image cleanup disabled, not deleting:", context.image_name)
 
-    context.add_cleanup(image_cleanup)
+    if not context.container_to_run:
+        context.add_cleanup(image_cleanup)
```

Comment:
I think you can move this context.add_cleanup up within the "else:" clause just below the line
_capture_container_as_image(build_container_name, context.image_name)

You can also move the def image_cleanup() up before your line
  if context.container_to_run:

That way you don't have to indent the declaration of def image_cleanup() any further in the

Reply:
right, thanks, I've changed it